Services/Sources/GitHubSource.cs

```diff
@@ -15,24 +15,51 @@
 * If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
 */
 
+using Models;
 using System;
 
 namespace Services.Sources
 {
-
+    /// <summary>
+    /// GitHubSource
+    /// </summary>
+    /// <seealso cref="Services.Sources.ISource" />
     public class GitHubSource : ISource
     {
-
+        /// <summary>
+        /// Gets the source.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <exception cref="NotImplementedException"></exception>
         public void GetSource(string url)
         {
             throw new NotImplementedException();
         }
+        /// <summary>
+        /// Gets the project information.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public Project GetProjectInformation(string url)
+        {
+            throw new NotImplementedException();
```

Comment:
This is because the frontend handles Github and i did not know if there was a need for the backend to have a github source

Reply:
I did not know that the front end was going to handle the GitHub URL's.
Why is this? What is the difference between a GitLab URL and a GitHub URL that makes us use the back end for one and the front end for the other? I thought that the back end was going to handle all URL's since the front end could be user-specific.

Reply:
This conversation was resolved, it might be worth adding the conclusion to this conversation before closing it so it's documented for future contributors.

Reply:
Yes. Please do! I would like to know what the outcome (and the reasoning behind it) is.

Reply:
the reason the frontend handles github is because when you want to support private projects it has to come from the frontend. storing refresh tokens is not a best practice.

Reply:
Does GitHub return private projects if we fetch the URL from our front end though? Since the sign-in session is created on GitHub, but the request comes from us.
This also means we need to do some URI filtering in the front end in order to distinguish. the GitHub URL's from all other URI's.
Anyway, the GitHub source can always be added in the future if necessary. Therefore this comment can be resolved.

Reply:
No private repositories are not yet supported, this will be made available next semester when we add the real "wizard" through the OAuth flow. In this wizard, users can select the projects they wish to import, private or public.

Reply:
It is possible to add the URL to a private repository (using the manual input). But I wonder if the fetch from the front end would be able to retrieve the private repository metadata as @wotwot563 said.